### ISSUE TITLE
server crashing after cloning a arena 3 times

### ISSUE THREAD

**Issue by serumanooo** [diamond-rip/Eden#343]
to be honest, i dont even know if its eden related, so here is what is happening and the console log:
https://youtu.be/zGmBadhj2fY
first i thought it was because i was cloning it too fast, but i tried to wait before cloning it again and it still happen like this, when i try to clone it for the third time, it crashes the server
[latest (5).log](https://github.com/diamond-rip/Eden/files/11617397/latest.5.log)

i dont think its related to the server, because its not using 100% of cpu and ram when i do that, the server just dies
but ill try it again with this option set to true
![image](https://github.com/diamond-rip/Eden/assets/132185379/cf6db54d-295e-41b0-b9e7-90679151dffd)

**Comment by serumanooo** [comment]
> 

i tried with "set-block-fast: true" but it still happens, ill see if its just with this arena or if its related to the quantity of clones (i dont think its it because there is a arena that i cloned like 10 times)

**Comment by serumanooo** [comment]
i was cloning other arenas and i found one that i just cant clone, it instantly kill the server

**Comment by GoodestEnglish** [comment]
It is related to Eden. I will investigate it more soon.

**Comment by HierarchyMan** [comment]
Never happened before but the server crashes when I try to clone any map for the first time. 
Blocks are still sometimes not resetting
Which is why you shouldn't allow players to play in the original arenas and always on a clone so you can paste the original arena over the clone to reset the arena without any issues.

**Comment by serumanooo** [comment]
> Blocks are still sometimes not resetting

This also happened to me, but idk if i saved the world  by accident or if its still not resetting

**Comment by serumanooo** [comment]
> @GoodestEnglish Never happened before but the server crashes when I try to clone any map for the first time. Blocks are still sometimes not resetting Which is why you shouldn't allow players to play in the original arenas and always on a clone so you can paste the original arena over the clone to reset the arena without any issues. My setblock fast is true Server software is papermc

yes, the arenas are still not reseting properly, i what @HierarchyMan suggested is a very good idea and it should fix all our problems

**Comment by GoodestEnglish** [comment]
> @GoodestEnglish Never happened before but the server crashes when I try to clone any map for the first time. Blocks are still sometimes not resetting Which is why you shouldn't allow players to play in the original arenas and always on a clone so you can paste the original arena over the clone to reset the arena without any issues. My setblock fast is true Server software is papermc

The arena generate task is kinda old. I am going to recode it soon. (The code is from Array, which is coded by zonix dev I believe back in 2017 lol)

I am still investigate the arena not reset issue. But this is kinda hard since I do not have this problem. If this keep happening I will make a dev version of Eden which disables all original arena and see if the problem still exists.

But as always, any error in console? (Even it is not related to Eden is fine) And what plugin do you have?

**Comment by HierarchyMan** [comment]
> But as always, any error in console? (Even it is not related to Eden is fine) And what plugin do you have?
> 
No errors the server just stops responding as soon as I click clone arena

**Comment by GoodestEnglish** [comment]
> > But as always, any error in console? (Even it is not related to Eden is fine) And what plugin do you have?
> 
> No errors the server just stops responding as soon as I click clone arena

I mean the arena reset bug. Is there any error in console when the arena resets/server starts? 
And what plugin do you have?

**Comment by HierarchyMan** [comment]
No errors
Might be a result of the server prematurely shutting down tbh

**Comment by GoodestEnglish** [comment]
> Might be a result of the server prematurely shutting down tbh

What do you mean by `Might be a result of the server prematurely shutting down tbh`? Do you mean the server shuts down when player is playing in your server?

**Comment by HierarchyMan** [comment]
> > Might be a result of the server prematurely shutting down tbh
> 
> What do you mean by `Might be a result of the server prematurely shutting down tbh`? Do you mean the server shuts down when player is playing in your server?

Yeah I host the server on my pc so I sometimes don't really /stop I just close the window

**Comment by GoodestEnglish** [comment]
> > > Might be a result of the server prematurely shutting down tbh
> > 
> > 
> > What do you mean by `Might be a result of the server prematurely shutting down tbh`? Do you mean the server shuts down when player is playing in your server?
> 
> Yeah I host the server on my pc so I sometimes don't really /stop I just close the window

I strongly (seriously) recommend not to do this. First of all this could break Eden's arena reset algorithm. Secondly this action could corrupt some unknown stuff and causes bugs.

**Comment by HierarchyMan** [comment]
Is this fixed?

**Comment by serumanooo** [comment]
> Is this fixed?

idk i havent tried to clone the sumo arenas again, but i think its something with the water, because only the sumo maps crash for me and they have water

**Comment by GoodestEnglish** [comment]
> Is this fixed?

Not yet. Since this requires a whole recode for the arena generate process, it needs to take some time.

> > Is this fixed?
> 
> idk i havent tried to clone the sumo arenas again, but i think its something with the water, because only the sumo maps crash for me and they have water

It is due to your arena coordinate. And yea this is going to be fixed soon
